src/core/oauth2-authorize.js

```diff
@@ -52,10 +52,17 @@ export default function authorize ( { auth, authActions, errActions, configs, au
   }
   query.push("redirect_uri=" + encodeURIComponent(redirectUrl))
 
-  if (Array.isArray(scopes) && 0 < scopes.length) {
+  let scopesArray = []
+  if (Array.isArray(scopes)) {
+    scopesArray = scopes
+  } else if (typeof scopes !== "undefined" && typeof scopes.toArray === "function") {
```

Comment:
Immutable has its own isType() static function which is more readable. Suggest importing Im, then change to `if (Im.List.isList(scopes)) { do something}`

Reply:
Yes, looks better